internal/update_test.go

```diff
@@ -58,3 +59,16 @@ func TestDetectUpdates(t *testing.T) {
 		}
 	}
 }
+
+func TestErrorAggregation(t *testing.T) {
+	viperInstance := viper.New()
+	viperInstance.SetConfigName(".git-ops-update")
+	viperInstance.SetConfigType("yaml")
+	viperInstance.AddConfigPath("../tests/testErrorHandling")
+	err := viperInstance.ReadInConfig()
+	assert.NoError(t, err)
+	config, err := LoadConfig(*viperInstance)
+	assert.NoError(t, err)
+	_, err = DetectUpdates("../tests/testErrorHandling", *config)
```

Comment:
Probably before running `DetectUpdates` the cache should be files with up-to-date data (or the test cases should all use invalid registry names). As it is right now, it looks like running this test would always make an actual call to https://registry-1.docker.io to get the current versions of `library/mariadb`

Reply:
@DizTortion there is now a `CacheProvider` interface with an existing `MemoryCacheProvider` implementation available in `cache_test.go` that can be used to easily mock the cache and so prevent actual requests when running `DetectUpdates`